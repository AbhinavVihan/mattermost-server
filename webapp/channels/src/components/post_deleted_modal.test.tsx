// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';

import PostDeletedModal from 'components/post_deleted_modal';
import {renderWithIntl} from 'tests/react_testing_utils';
import {fireEvent, screen} from '@testing-library/react';

describe('components/ChannelInfoModal', () => {
    const baseProps = {
        onExited: jest.fn(),
    };

    test('should match snapshot', () => {
        const {baseElement} = renderWithIntl(
            <PostDeletedModal {...baseProps}/>,
        );

        fireEvent.click(screen.getByTestId('postDeletedModalOkButton'));
        expect(baseProps.onExited).toHaveBeenCalledTimes(1);

        expect(baseElement).toMatchSnapshot();
    });
});
