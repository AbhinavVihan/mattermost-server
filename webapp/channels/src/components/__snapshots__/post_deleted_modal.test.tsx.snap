// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ChannelInfoModal should match snapshot 1`] = `
<body
  class="modal-open"
  style="padding-right: 0px;"
>
  <div
    aria-hidden="true"
  />
  <div
    role="dialog"
  >
    <div
      class="fade modal-backdrop"
    />
    <div
      aria-labelledby="postDeletedModalLabel"
      class="fade modal"
      data-testid="postDeletedModal"
      role="dialog"
      style="display: block; padding-right: 0px;"
      tabindex="-1"
    >
      <div
        class="a11y__modal modal-dialog"
      >
        <div
          class="modal-content"
          role="document"
        >
          <div
            class="modal-header"
          >
            <button
              aria-label="Close"
              class="close"
              type="button"
            >
              <span
                aria-hidden="true"
              >
                Ã—
              </span>
              <span
                class="sr-only"
              >
                Close
              </span>
            </button>
            <h1
              class="modal-title"
              id="postDeletedModalLabel"
            >
              Comment could not be posted
            </h1>
          </div>
          <div
            class="modal-body"
          >
            <p>
              Someone deleted the message on which you tried to post a comment.
            </p>
          </div>
          <div
            class="modal-footer"
          >
            <button
              class="btn btn-primary"
              data-testid="postDeletedModalOkButton"
              type="button"
            >
              Okay
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
